### PR TITLE
add "messy" station trait; add party trash to hangover trait

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR adds a new station trait, "messy", which generates several distinct kinds of messes (consisting of decals and relevant objects) on roundstart. It also adds a special "party" mess which spawns when the hangover trait is rolled.
## Why It's Good For The Game
Seems in the spirit of station traits, gives janitors something to do early round instead of waiting for messes to occur.
## Images of changes
For the station trait itself, there are the following mess kinds:
### Bloody
Bloody messes choose a species color, as well as drop various body parts.
2024_03_07__08_34_04__Paradise Station 13

### Robotic
Robotic messes use the "machine blood" color, and drop robot gibs.
2024_03_07__07_01_42__Paradise Station 13

### Engineering
Engineering messes burn and break tiles, and drop glass shreds, tiny pieces of cable, rods, and tiles. Note it does not actually disconnect any cables, just drops small pieces of cable.

This is more for engineers to fix up than janitors.
2024_03_07__06_23_14__Paradise Station 13

### Dirty
Dirty messes add dirt, insect guts, and other miscellaneous decals.
2024_03_07__07_01_06__Paradise Station 13

### Cooking
Cooking messes cause flour, cooking oil, ants, and egg white decals, and drop plates, meat, and snackbowls.
2024_03_07__06_56_24__Paradise Station 13

### Party
When the hangover trait is rolled, party messes are generated, which add confetti, vomit, cigarette butts, and broken bottles.
2024_03_07__06_47_48__Paradise Station 13

## Testing
Repeated spawning in and checking placement and layout of generated messes.

## Changelog
:cl:
add: New station trait: messy station. The last shift didn't clean up after themselves!
add: Hangover station trait now includes appropriately festive trash.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/spawners/mess_spawners.dm [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+		/obj/effect/decal/cleanable/blood,
+		/obj/effect/decal/cleanable/blood/splatter,
+		/obj/effect/decal/cleanable/blood/drip,
+		/obj/effect/decal/cleanable/blood/gibs,
+	)
+	var/obj/effect/decal/cleanable/blood/decal = new blood_decal_type(T)
+	decal.basecolor = bloodcolor
+	decal.update_icon()
+
+	if(prob(5))
```
Comment (1080pCat) [inline code/game/objects/effects/spawners/mess_spawners.dm:99]: just a fwi, im fairly sure these aren't cleanable/removable. iirc the robotic ones are, but only by the drones

Comment (1080pCat) [reply code/game/objects/effects/spawners/mess_spawners.dm:99]: me when wrong line
line 100, the remains, skull and crossbones thing
image

Comment (SynthTwo) [reply code/game/objects/effects/spawners/mess_spawners.dm:99]: You can't clean remains, yeah, it sucks

**Comment by Probablyjay19** [comment]
Ah yes, what a night on the town. I was so hyped when yaya pecked that dude's eyes clean out of his head and the chef made a spaghetti slip and slide all over central primary. Oh, and when the captain ultrakilled that borg and turned his body into a fountain? Good times.

**Review by AffectedArc07 — APPROVED** [review]

File: code/game/objects/effects/spawners/mess_spawners.dm (hunk 1 above)
Comment (warriorstar-orion) [reply code/game/objects/effects/spawners/mess_spawners.dm]: sad. removed this for now.

File: code/game/objects/effects/spawners/mess_spawners.dm [hunk 2]
```diff
@@ -0,0 +1,203 @@
+/obj/effect/spawner/themed_mess
+	name = "mess spawner"
+	icon = 'icons/effects/spawner_icons.dmi'
+	icon_state = "questionmark"
+
+	var/remaining_tiles
```
Comment (lewcc) [inline code/game/objects/effects/spawners/mess_spawners.dm:6]: Autodoc this please

**Review by lewcc — COMMENTED** [review]
One quick nit

File: code/game/objects/effects/spawners/mess_spawners.dm (hunk 2 above)
Comment (warriorstar-orion) [reply code/game/objects/effects/spawners/mess_spawners.dm:9]: Added.

**Review by lewcc — APPROVED** [review]

**Pull request merged**